pkg/postprocessors/retry_test.go

```diff
@@ -0,0 +1,40 @@
+package postprocessors
+
+import (
+	"context"
+	"testing"
+)
+
+type mockRetryer struct {
+	retryCalled bool
+}
+
+func (m *mockRetryer) Retry(result CheckResult) {
+	m.retryCalled = true
+}
+
+func TestRetryPostProcessor_PostProcess(t *testing.T) {
+	// Create a mock retryer
+	retryer := &mockRetryer{}
+
+	// Create a RetryPostProcessor with the mock retryer
+	processor := NewRetryPostProcessor(retryer)
+
+	// Create some check results
+	results := []CheckResult{
+		{Retryable: true},
+		{Retryable: false},
+		{Retryable: true},
+	}
+
+	// Call the PostProcess method
+	err := processor.PostProcess(context.Background(), results)
+	if err != nil {
```

Comment:
Instead of using checks like `if`, I think it's cleaner in tests to use `assert`, e.g. you can replace this if block with:

assert.Nil(t, err, "PostProcess returned an error: %v", err)